interfaces/prompting/requestprompts/requestprompts_test.go

```diff
@@ -1070,3 +1081,299 @@ func (s *requestpromptsSuite) TestPromptMarshalJSON(c *C) {
+		allowedFilePermission, ok := allowedPermission.(notify.FilePermission)
+		c.Assert(ok, Equals, true)
+		replyChan <- allowedFilePermission
+		return nil
+	})
+	defer restore()
+
+	metadata := &prompting.Metadata{
+		User:      s.defaultUser,
+		Snap:      "firefox",
+		Interface: "home",
+	}
+	path := "/home/test/foo"
+	requestedPermissions := []string{"read", "write", "execute"}
+	remainingPermissions := []string{"write", "execute"}
+
+	noticeChan := make(chan noticeInfo, 1)
+	pdb, err := requestprompts.New(func(userID uint32, promptID prompting.IDType, data map[string]string) error {
+		c.Assert(userID, Equals, s.defaultUser)
+		noticeChan <- noticeInfo{
+			promptID: promptID,
+			data:     data,
+		}
+		return nil
+	})
+	c.Assert(err, IsNil)
+	defer pdb.Close()
+
+	// Add prompt
+	listenerReq := &listener.Request{}
+	prompt, merged, err := pdb.AddOrMerge(metadata, path, requestedPermissions, remainingPermissions, listenerReq)
+	c.Assert(err, IsNil)
+	c.Assert(merged, Equals, false)
+	checkCurrentNotices(c, noticeChan, prompt.ID, nil)
+
+	// Check that prompt has not immediately expired
+	c.Assert(timer.FireCount(), Equals, 0)
+
+	// Cause prompt to timeout, but the callback will wait for a signal, so we
+	// can reset it, simulating activity occurring just as the timer fires.
+	timer.Elapse(requestprompts.InitialTimeout)
+
+	// Check that the timer fired
+	c.Assert(timer.FireCount(), Equals, 1)
+
+	// Reset timer to half of initial timeout, as if activity occurred, but it's
+	// easier to check that the timeout was correctly reset to activityTimeout
+	// if the preemptive reset was not also to activityTimeout.
+	//
+	// In the real world, what would have happened is that activity occurred
+	// just as the timer fired, thus resetting the timer to activityTimeout
+	// just before the timeout callback sets it to initialTimeout, and we want
+	// to ensure that the callback correctly detects that the activity had
+	// occurred (by the timer being active again) and overrides its own just-set
+	// initialTimeout by resetting the timer back to activityTimeout.
+	timer.Reset(requestprompts.InitialTimeout / 2)
+
+	// Start the actual callback
+	callbackSignaller <- true
+	// Wait for the callback to complete
+	<-callbackSignaller
+
+	// Check that prompt has not expired
+	clientActivity := false
+	retrieved, err := pdb.PromptWithID(s.defaultUser, prompt.ID, clientActivity)
+	c.Assert(err, IsNil)
+	c.Assert(retrieved, Equals, prompt)
+	c.Assert(timer.FireCount(), Equals, 1)
+
+	// Check that the callback correctly identified that the timer had been
+	// reset prior to the callback doing so, and thus re-reset the timeout to
+	// activityTimeout instead of leaving it reset to initialTimeout.
+	// First, check that the prompt doesn't expire after the preemptively reset
+	// timeout (but before initialTimeout).
+	timer.Elapse(requestprompts.InitialTimeout / 4 * 3)
```

Comment:
Same comment to use the expected amount and keep subtracting from it so that we are not at the mercy of equal divisibility by accident.